src/sentry/integrations/slack/endpoints/action.py

```diff
@@ -84,6 +84,61 @@ def update_group(
     )
 
 
+def get_group(slack_request: SlackActionRequest) -> Group | None:
+    """Determine the issue group on which an action is being taken."""
+    group_id = slack_request.callback_data["issue"]
+    try:
+        return Group.objects.select_related("project__organization").get(
+            id=group_id,
+            project__organization__organizationintegration__integration=slack_request.integration,
+        )
+    except Group.DoesNotExist:
+        logger.info(
+            "slack.action.invalid-issue",
+            extra={
+                **slack_request.logging_data,
+                "group_id": group_id,
+            },
+        )
+        return None
+
+
+def get_member_approval_success_message_parts(
```

Comment:
@mgaeta can we return an instance of a datclass instead? Would be nice to have a more structured response so we can get proper type checking on field names. Concerned about bugs where not every field in `kwargs` is set in every code path.